### ISSUE TITLE
Wrong normals of built-in torus shape

### ISSUE THREAD

**Issue by YoshieraHuang** [bevyengine/bevy#4896]
## Bevy version

"0.7"

## Relevant system information

```ignore
`AdapterInfo { name: "Radeon RX 590 Series", vendor: 4098, device: 26591, device_type: DiscreteGpu, backend: Vulkan }`
```

## What you did

I try to build my outline shader using [vertex extrusion method](https://www.videopoetics.com/tutorials/pixel-perfect-outline-shaders-unity). This method is strongly related to the normals of vertices. 

## What went wrong

I tests many meshes and my code does well expect the built-in torus shape.
This is the result of torus imported from Blender and it looks good:

![blender_torus](https://user-images.githubusercontent.com/38752027/171549213-dc6eefea-d864-4f1a-be6d-a66241c55931.png)

This is the result of built-in torus and it looks not so good:

![bevy_torus](https://user-images.githubusercontent.com/38752027/171549390-931bbbd8-e330-47a3-9299-29c67a8a6166.png)

After carefully testing, I find that some of the calculated normals of built-in torus may be wrong.

I will check the code later and find what goes wrong actually.

**Comment by hymm** [comment]
this should be fixed on main in https://github.com/bevyengine/bevy/pull/4520.

**Comment by YoshieraHuang** [comment]
> this should be fixed on main in #4520.

Thank you. This also fixes my problem. I will close this issue
